### ISSUE TITLE
Error when replicating Conformer encoder + LSTM decoder results for librispeech

### ISSUE THREAD

**Issue by usoylu2** [k2-fsa/icefall#1088]
When I try to reproduce the training for Conformer encoder + LSTM decoder at https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/RESULTS.md via  

```
export CUDA_VISIBLE_DEVICES="0,1,2,3"

./transducer/train.py \
  --world-size 4 \
  --num-epochs 35 \
  --start-epoch 0 \
  --exp-dir transducer/exp-lr-2.5-full \
  --full-libri 1 \
  --max-duration 180 \
  --lr-factor 2.5
```

I get this error: 

Traceback (most recent call last):                                         
  File "./transducer/train.py", line 748, in <module>                     
    main()                                                                 
  File "./transducer/train.py", line 739, in main                          
    mp.spawn(run, args=(world_size, args), nprocs=world_size, join=True)     
  File "/local/home/ufsoylu/my_env/lib64/python3.7/site-packages/torch/mult
iprocessing/spawn.py", line 240, in spawn                                   
    return start_processes(fn, args, nprocs, join, daemon, start_method='sp   
awn')                                                                       
  File "/local/home/ufsoylu/my_env/lib64/python3.7/site-packages/torch/mult  
iprocessing/spawn.py", line 198, in start_processes                       
    while not context.join():                                             
  File "/local/home/ufsoylu/my_env/lib64/python3.7/site-packages/torch/mult
iprocessing/spawn.py", line 160, in join                                  
    raise ProcessRaisedException(msg, error_index, failed_process.pid)    
torch.multiprocessing.spawn.ProcessRaisedException:                          
                                                                           
-- Process 3 terminated with the following error:                          
Traceback (most recent call last):                                       
  File "/local/home/ufsoylu/my_env/lib64/python3.7/site-packages/torch/mult  
iprocessing/spawn.py", line 69, in _wrap                                   
    fn(i, *args)                                             
File "/local/home/ufsoylu/icefall_tmp/icefall/egs/librispeech/ASR/transdu
cer/train.py", line 630, in run                                          
    num_left = len(train_cuts)                                        
  File "/local/home/ufsoylu/my_env/lib64/python3.7/site-packages/lhotse/cut  
/set.py", line 2571, in _ _len_ _                                        
    return len(self.cuts)                                                 
  File "/local/home/ufsoylu/my_env/lib64/python3.7/site-packages/lhotse/laz 
y.py", line 390, in _ _len_ _                                         
    "LazyFilter does not support _ _len_ _ because it would require "        
NotImplementedError: LazyFilter does not support _ _len_ _ because it would r   
equire iterating over the whole iterator, which is not possible in a lazy f
ashion. If you really need to know the length, convert to eager mode first     
using `.to_eager()`. Note that this will require loading the whole iterator
 into memory.

**Comment by csukuangfj** [comment]
Will fix it tomorrow.

By the way, the folder transducer is quite old. Please use ./zipformer.

**Comment by usoylu2** [comment]
Thanks!

**Comment by csukuangfj** [comment]
Fixed in #1096
